controllers/dockermachine_controller.go

```diff
@@ -91,130 +123,70 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	}
-	state := getState(machine, dockerMachine)
-	log = log.WithValues("state", state.String())
-	// TODO: consider setting the key pieces in this function, such as ProviderID and deleting finalizers
-	switch state {
-	case Deleted:
-		log.Info("Start reconcileDelete dockerMachine")
-		return r.reconcileDelete(ctx, cluster, machine, dockerMachine)
-	case Provisioned:
-		return ctrl.Result{}, nil
-	case Pending:
-		log.Info("Waiting for machine bootstrap")
-		return ctrl.Result{}, nil
-	case Provisioning:
-		log.Info("Provisioning machine")
-		// Ensuring cluster is ready for joining
-		clusterExists, err := kindcluster.IsKnown(cluster.Name)
-		if err != nil {
-			log.Error(err, "Error finding cluster-name", "cluster", cluster.Name)
-			return ctrl.Result{}, err
-		}
 
-		// If there's no cluster, requeue the request until there is one
-		if !clusterExists {
-			r.Log.Info("There is no cluster yet, waiting for a cluster before creating machines")
-			return ctrl.Result{RequeueAfter: time.Second * 30}, nil
-		}
-		log.Info("Creating machine")
-		return r.create(ctx, cluster, machine, dockerMachine)
-	default:
-		log.Info("Unknown state", "state", state)
+	// if the machine is already provisioned, return
+	if dockerMachine.Spec.ProviderID != nil {
 		return ctrl.Result{}, nil
 	}
-}
-
-// State is the state our machine object is in
-type State int
 
-const (
-	// Provisioned is the state that happens after the machine is part of a cluster.
-	// Technically it should be Running but this controller sees no difference between the two.
-	Provisioned State = iota
-	// Pending is when the machine is waiting for bootstrap data to exist.
-	Pending
-	// Deleted is when the machine has been deleted.
-	Deleted
-	// Provisioning is when bootstrap data exists but it's not finished being provisioned.
-	Provisioning
-)
-
-// String is a helper function to print a human-readable form of the state.
-func (s State) String() string {
-	switch s {
-	case Provisioning:
-		return "Provisioning"
-	case Deleted:
-		return "Deleted"
-	case Provisioned:
-		return "Provisioned"
-	case Pending:
-		return "Pending"
-	default:
-		return "Unknown"
+	// Make sure infrastructure is ready
+	if !machinesvc.Cluster.Status.InfrastructureReady {
+		machinesvc.Info("Cluster infrastructure is not ready yet, requeuing")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
-}
 
-func getState(machine *clusterv1.Machine, dockerMachine *infrav1.DockerMachine) State {
-	// Deleted takes precedence
-	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
-		return Deleted
-	}
-	if dockerMachine.Spec.ProviderID != nil {
-		return Provisioned
-	}
-	if machine.Spec.Bootstrap.Data == nil {
-		return Pending
+	// Make sure bootstrap data is available and populated.
+	if machinesvc.Machine.Spec.Bootstrap.Data == nil {
+		machinesvc.Info("Waiting for bootstrap data to be available")
+		return ctrl.Result{}, nil
```

Comment:
```suggestion
		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Reply:
No, we should not requeue here, but rather wait for the watch on Machines to trigger the reconcile.